velox/functions/sparksql/coverage/Coverage.cpp

```diff
@@ -0,0 +1,380 @@
+  return names;
+}
+
+/// Returns alphabetically sorted list of aggregate functions available in
+/// Velox.
+std::vector<std::string> getSortedAggregateNames() {
+  const auto& functions = exec::aggregateFunctions();
+
+  std::vector<std::string> names;
+  names.reserve(functions.size());
+  for (const auto& entry : functions) {
+    names.push_back(entry.first);
+  }
+  std::sort(names.begin(), names.end());
+  return names;
+}
+
+void printVeloxFunctions() {
+  // Do not add links for arithmetic operators.
+  static const std::unordered_set<std::string> kLinkBlockList = {"in"};
+
+  auto scalarNames = getSortedScalarNames();
+  auto aggNames = getSortedAggregateNames();
+
+  const int columnSize = std::max(maxLength(scalarNames), maxLength(aggNames)) +
+      toFuncLink("").size();
+
+  const std::string indent(4, ' ');
+
+  auto scalarCnt = scalarNames.size();
+  auto aggCnt = aggNames.size();
+  auto numRows = std::max((size_t)std::ceil(scalarCnt / 3.0), aggCnt);
+
+  auto printName = [&](const std::string& name) {
+    return kLinkBlockList.count(name) == 0 ? toFuncLink(name) : name;
+  };
+
+  TablePrinter printer(3, columnSize, indent, std::cout);
+  printer.header();
+  for (int i = 0; i < numRows; i++) {
+    printer.startRow();
+
+    // 3 columns of scalar functions.
+    for (int j = 0; j < 3; j++) {
+      auto n = i + numRows * j;
+      n < scalarCnt ? printer.addColumn(printName(scalarNames[n]))
+                    : printer.addEmptyColumn();
+    }
+
+    // 1 empty column.
+    printer.addEmptyColumn();
+
+    // 1 column of aggregate functions.
+    i < aggCnt ? printer.addColumn(printName(aggNames[i]))
+               : printer.addEmptyColumn();
+
+    printer.endRow();
+  }
+  printer.footer();
+}
+
+/// Takes a super-set of simple, vector and aggregate function names and prints
+/// coverage map showing which of these functions are available in Velox.
+void printCoverageMap(
+    const std::vector<std::string>& scalarNames,
+    const std::vector<std::string>& aggNames) {
+  auto veloxFunctions = getFunctionSignatures();
+
+  std::unordered_set<std::string> veloxNames;
+  veloxNames.reserve(veloxFunctions.size());
+  for (const auto& func : veloxFunctions) {
+    veloxNames.emplace(func.first);
+  }
+
+  std::unordered_set<std::string> veloxAggNames;
+
+  const auto& veloxAggregateFunctions = exec::aggregateFunctions();
+  for (const auto& entry : veloxAggregateFunctions) {
+    veloxAggNames.emplace(entry.first);
+  }
+
+  printCoverageMap(scalarNames, aggNames, veloxNames, veloxAggNames);
+}
+
+void printCoverageMapForAll() {
+  auto scalarNames = readFunctionNamesFromFile("all_scalar_functions.txt");
+  std::sort(scalarNames.begin(), scalarNames.end());
+
+  auto aggNames = readFunctionNamesFromFile("all_aggregate_functions.txt");
+  std::sort(aggNames.begin(), aggNames.end());
+
+  printCoverageMap(scalarNames, aggNames);
+}
+
+} // namespace
+
+int main(int argc, char** argv) {
```

Comment:
Is this a copy-paste of the existing Coverage.cpp?  If so, let's refactor to avoid code duplication.

Reply:
I will refactor to avoid code duplication.